resources/schema.gql

```diff
@@ -286,6 +294,25 @@ type DatasetEdge {
 	node: Dataset!
 }
 
+type DatasetFlowConfigs {
+	"""
+	Returns defined configuration for a flow of specified type
+	"""
+	flowConfig(datasetFlowType: DatasetFlowType!): FlowConfiguration
+}
+
+type DatasetFlowConfigsMut {
+	setConfigTimeDelta(datasetFlowType: DatasetFlowType!, paused: Boolean!, every: TimeDeltaInput!): FlowConfiguration!
+	setConfigCronExpression(datasetFlowType: DatasetFlowType!, paused: Boolean!, cronExpression: String!): FlowConfiguration!
+	setConfigBatching(datasetFlowType: DatasetFlowType!, paused: Boolean!, throttlingPeriod: TimeDeltaInput, minimalDataBatch: Int): FlowConfiguration!
+}
```

Comment:
This seems to be begging for a `union` too, but on the input side.

I found this [closed discussion](https://github.com/graphql/graphql-spec/issues/488) about adding `inputUnion` support.

They recommend [`@oneOf`](https://github.com/graphql/graphql-spec/pull/825) as an alternative - have you checked if `async-graphql` supports it?

Reply:
I tried, and surprisingly it worked. My main concern was more of TypeScript side.
It looks weird enough both in schema:

```
input ScheduleInput @oneOf {
	timeDelta: TimeDeltaInput
	cronExpression: String
}
```

and in TypeScript:

```
export type ScheduleInput =
    | { cronExpression: Scalars["String"]; timeDelta?: never }
    | { cronExpression?: never; timeDelta: TimeDeltaInput };
```

but it works fine:

```
const schedule_1: ScheduleInput = { timeDelta: { every: 1, unit: TimeUnit.Weeks } };
const schedule_2: ScheduleInput = { cronExpression: "* */2 * * *" };
```

and event type checking is fine - you can't write both or none.

Reply:
So, with this weapon, I decided to have 2 instead of 3 methods - I still prefer explicit separation for root and derived datasets, as these would be differently created forms in UI:

```
type DatasetFlowConfigsMut {
	setConfigSchedule(datasetFlowType: DatasetFlowType!, paused: Boolean!, schedule: ScheduleInput!): FlowConfiguration!
	setConfigBatching(datasetFlowType: DatasetFlowType!, paused: Boolean!, throttlingPeriod: TimeDeltaInput, minimalDataBatch: Int): FlowConfiguration!
}
```